src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TransitionBlock.cs

```diff
@@ -73,6 +75,22 @@ public static TransitionBlock FromTarget(TargetDetails target)
 
         public abstract int SizeOfTransitionBlock { get; }
 
+        public int SizeOfPInvokeTransitionBlock => SizeOfGSCookie + SizeOfInlinedTransitionFrame;
+
+        private int SizeOfGSCookie => PointerSize;
+
+        private int SizeOfInlinedTransitionFrame => SizeOfFrame + SizeOfInlinedCallFrame; // base class size + inlined call frame fields
```

Comment:
A nit - this seems too verbose and a bit hard to read to me. How about this instead?
```C#
private int SizeOfInlinedTransitionFrame => 
    2 * PointerSize + // Base class - vtable, m_Next
    5 * PointerSize + // m_Datum, m_pCallSiteSP, m_pCallerReturnAddress, m_pCalleeSavedFP, m_pThread
    (Is64Bit ? 1 : 0) * PointerSize + // m_StubSecretArg
    (IsARM || IsARM64 ? 1 : 0) * PointerSize; // m_pSPAfterProlog
    
```